shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.mm

```diff
@@ -453,13 +461,12 @@ - (void)testWillDeallocNotification {
     FlutterViewController* realVC = [[FlutterViewController alloc] initWithEngine:engine
                                                                           nibName:nil
                                                                            bundle:nil];
-    id observer =
-        [[NSNotificationCenter defaultCenter] addObserverForName:FlutterViewControllerWillDealloc
-                                                          object:nil
-                                                           queue:[NSOperationQueue mainQueue]
-                                                      usingBlock:^(NSNotification* _Nonnull note) {
-                                                        [expectation fulfill];
-                                                      }];
+    [[NSNotificationCenter defaultCenter] addObserverForName:FlutterViewControllerWillDealloc
```

Comment:
You can use https://developer.apple.com/documentation/xctest/xctestcase/1500964-expectationfornotification?language=objc to fulfill on notifications.